### PR TITLE
[11.0] [MIG] Hotel Restaurant

### PR DESCRIPTION
**Pull request by serpentcs-dev1**
Hotel Restaurant
=============

You can manage:

* **Restaurant Configuration (Food Item, Table)**

* **Table Reservation**

* **Manage Food Orders**

* **Manage Kitchen Order ticket**

* **Quick Table Reservation Facility With Its Food Order**

* **Order's History In Hotel Folio**

* **Payment**

* **Different Reports are also provided, mainly for Restaurant.**

### CODE REVIEW

File: hotel_restaurant/README.rst [hunk 1]
```diff
@@ -0,0 +1,80 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: https://www.gnu.org/licenses/agpl
+   :alt: License: AGPL-3
+
+=================
+Hotel Restaurant
+=================
```
Comment (Trivedi-Vacha-SerpentCS) [inline hotel_restaurant/README.rst]: Align of star with t

File: hotel_restaurant/README.rst [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+* Payment
+
+* Different Reports are also provided, mainly for Restaurant.
+
+Usage
+=====
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/157/10.0
```
Comment (Trivedi-Vacha-SerpentCS) [inline hotel_restaurant/README.rst]: Check version, should 11.0

File: hotel_restaurant/README.rst [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+This Module is providing table booking facilities and Managing customers orders
+
+You can manage:
+
+* Restaurant Configuration (Food Item, Table)
+
+* **Table Reservation**
+
+.. image:: https://raw.githubusercontent.com/OCA/vertical-hotel/10.0/hotel_restaurant/static/description/res2.png
```
Comment (Trivedi-Vacha-SerpentCS) [inline hotel_restaurant/README.rst]: Check path of an image it should be 11.0 instead of 10.0

File: hotel_restaurant/README.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+check there if your issue has already been reported. If you spotted it first,
+help us smash it by providing detailed and welcomed feedback.
+
+Credits
+=======
+
+Contributors
+------------
+
+* Serpent Consulting Services Pvt. Ltd. <jay.vora@serpentcs.com> (http://www.serpentcs.com)
```
Comment (Trivedi-Vacha-SerpentCS) [inline hotel_restaurant/README.rst]: no need of website

File: hotel_restaurant/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# See LICENSE file for full copyright and licensing details.
+
+{
+    'name': 'Hotel Restaurant Management',
+    'version': '11.0.1.0.0',
+    'author': 'Odoo Community Association (OCA), Serpent Consulting\
+                Services Pvt. Ltd., Odoo S.A.',
+    'category': 'Generic Modules/Hotel Restaurant',
+    'website': 'http://www.serpentcs.com',
```
Comment (Trivedi-Vacha-SerpentCS) [inline hotel_restaurant/__manifest__.py]: Website to be `https://github.com/OCA/vertical-hotel/`

File: hotel_restaurant/models/hotel_restaurant.py [hunk 6]
```diff
@@ -0,0 +1,728 @@
...
+
+
+class HotelMenucardType(models.Model):
+
+    _name = 'hotel.menucard.type'
+    _description = 'Food Item Type'
+
+    name = fields.Char('Name', required=True)
+    menu_id = fields.Many2one('hotel.menucard.type', string='Food Item Type')
+    child_id = fields.One2many('hotel.menucard.type', 'menu_id',
```
Comment (Trivedi-Vacha-SerpentCS) [inline hotel_restaurant/models/hotel_restaurant.py]: child_ids- field name

File: hotel_restaurant/models/hotel_restaurant.py [hunk 7]
```diff
@@ -0,0 +1,728 @@
...
+        """
+        This method is for create a new order for hotel restaurant
+        reservation .when table is booked and create order button is
+        clicked then this method is called and order is created.you
+        can see this created order in "Orders"
+        ------------------------------------------------------------
+        @param self: The object pointer
+        @return: new record set for hotel restaurant reservation.
+        """
+        proxy = self.env['hotel.reservation.order']
```
Comment (Trivedi-Vacha-SerpentCS) [inline hotel_restaurant/models/hotel_restaurant.py]: please improve variable name

File: hotel_restaurant/models/hotel_restaurant.py [hunk 8]
```diff
@@ -0,0 +1,728 @@
...
+    @api.onchange('folio_id')
+    def get_folio_id(self):
+        '''
+        When you change folio_id, based on that it will update
+        the cname and room_number as well
+        ---------------------------------------------------------
+        @param self: object pointer
+        '''
+        for rec in self:
+            self.cname = False
```
Comment (Trivedi-Vacha-SerpentCS) [inline hotel_restaurant/models/hotel_restaurant.py]: IMO it should be rec.cname = False

File: hotel_restaurant/models/hotel_restaurant.py [hunk 9]
```diff
@@ -0,0 +1,728 @@
...
+            res = self._cr.fetchone()
+            roomcount = res and res[0] or 0.0
+            if len(reservation.tableno.ids) == 0:
+                raise ValidationError(_('Please Select Tables For \
+                                         Reservation'))
+            if roomcount:
+                raise ValidationError(_('You tried to confirm reservation \
+                                         with table those already reserved \
+                                         in this reservation period'))
+            else:
```
Comment (Trivedi-Vacha-SerpentCS) [inline hotel_restaurant/models/hotel_restaurant.py]: IMO  We don't need else

File: hotel_restaurant/models/hotel_restaurant.py [hunk 10]
```diff
@@ -0,0 +1,728 @@
...
+        return True
+
+    _name = "hotel.reservation.order"
+    _description = "Reservation Order"
+    _rec_name = "order_number"
+
+    order_number = fields.Char('Order No', readonly=True)
+    reservationno = fields.Many2one('hotel.restaurant.reservation',
+                                    'Reservation No')
+    date1 = fields.Datetime('Date', required=True,
```
Comment (Trivedi-Vacha-SerpentCS) [inline hotel_restaurant/models/hotel_restaurant.py]: field name should be date_order of meaning full

File: hotel_restaurant/models/hotel_restaurant.py [hunk 11]
```diff
@@ -0,0 +1,728 @@
...
+                            default=(lambda *a:
+                                     time.strftime
+                                     (DEFAULT_SERVER_DATETIME_FORMAT)))
+    waitername = fields.Many2one('res.partner', 'Waiter Name')
```
Comment (Trivedi-Vacha-SerpentCS) [inline hotel_restaurant/models/hotel_restaurant.py]: should end with _id

File: hotel_restaurant/models/hotel_restaurant.py [hunk 12]
```diff
@@ -0,0 +1,728 @@
...
+        for order in self:
+            table_ids = [x.id for x in order.table_no]
+            line_data = {
+                'orderno': order.order_number,
+                'resno': order.reservationno.reservation_id,
+                'kot_date': time.strftime(DEFAULT_SERVER_DATETIME_FORMAT),
+                'w_name': order.waitername.name,
+                'tableno': [(6, 0, table_ids)],
+                }
+            kot_obj = order_tickets_obj.browse(self.kitchen_id)
```
Comment (Trivedi-Vacha-SerpentCS) [inline hotel_restaurant/models/hotel_restaurant.py]: should be kot_rec

File: hotel_restaurant/models/hotel_restaurant.py [hunk 13]
```diff
@@ -0,0 +1,728 @@
...
+            kot_obj.write(line_data)
+            for order_line in order.order_list:
+                if order_line.id not in order.rest_id.ids:
+                    kot_data1 = order_tickets_obj.create(line_data)
+                    self.kitchen_id = kot_data1.id
```
Comment (Trivedi-Vacha-SerpentCS) [inline hotel_restaurant/models/hotel_restaurant.py]: should be order.kitchen_id

File: hotel_restaurant/models/hotel_restaurant.py [hunk 14]
```diff
@@ -0,0 +1,728 @@
...
+        ----------------------------------------
+        @param self: object pointer
+        """
+        hotel_folio_obj = self.env['hotel.folio']
+        hsl_obj = self.env['hotel.service.line']
+        so_line_obj = self.env['sale.order.line']
+        for order_obj in self:
+            hotelfolio = order_obj.folio_id.order_id.id
+            if order_obj.folio_id:
+                for order1 in order_obj.order_list:
```
Comment (Trivedi-Vacha-SerpentCS) [inline hotel_restaurant/models/hotel_restaurant.py]: should be `order`

**Review by Trivedi-Vacha-SerpentCS — CHANGES_REQUESTED** [review]

**Comment by serpentcs-dev1** [comment]
@Trivedi-Vacha-SerpentCS Appreciate the Review, let me amend the changes!

**Comment by serpentcs-dev1** [comment]
@Trivedi-Vacha-SerpentCS The requested changes have been made, can you please approve your review?!

**Review by Trivedi-Vacha-SerpentCS — APPROVED** [review]

**Review by Trivedi-Vacha-SerpentCS — APPROVED** [review]

**Pull request merged**